src/qsurvey/__init__.py

```diff
@@ -191,14 +218,14 @@ def __init__(
             schedule,
             [undesirable_courses],
             [0],
-            course,
+            [course, section],
             sparse,
         )
         self.preferred_courses_constraint = PreferenceConstraint.from_item_lists(
             schedule,
-            [self.preferred_courses],
+            [[self.preferred_courses]],
```

Comment:
@gbiss in the fair module, this usually takes the preferred courses divided by topic. In this case, we just have the preferred courses, so there is just a list of ScheduleItems instead of a list of lists of ScheduleItems. I think this quick fix is probably ok, but would like to know if you have concerns about it.